2003-12-15  Dmitry V. Levin <ldv@altlinux.org>

	* strace.c [WCOREDUMP](trace): Show coredump status of the
	killed process if available.

--- strace-4.5.1.orig/strace.c	2003-11-12 00:24:23 +0300
+++ strace-4.5.1/strace.c	2003-12-15 00:39:26 +0300
@@ -2022,8 +2022,12 @@ Process %d attached (waiting for parent)
 			if (!cflag
 			    && (qual_flags[WTERMSIG(status)] & QUAL_SIGNAL)) {
 				printleader(tcp);
-				tprintf("+++ killed by %s +++",
-					signame(WTERMSIG(status)));
+				tprintf("+++ killed by %s %s+++",
+					signame(WTERMSIG(status)),
+#ifdef WCOREDUMP
+					WCOREDUMP(status) ? "(core dumped) " :
+#endif
+					"");
 				printtrailer(tcp);
 			}
 #ifdef TCB_GROUP_EXITING
